lang/cs/Org.Apache.REEF.Wake/Remote/IDataReader.cs

```diff
@@ -143,5 +144,12 @@ public interface IDataReader
         /// <returns>Task handler that reads bytes and returns that number of bytes read 
         /// if success, otherwise -1</returns>
         Task<int> ReadAsync(byte[] buffer, int index, int bytesToRead, CancellationToken token);
+
+        /// <summary>
+        /// Gets underlying stream. Throws null exception if 
+        /// stream is null or not available
+        /// </summary>
+        /// <returns>The underlying stream</returns>
+        Stream GetStream();
```

Comment:
If the class is thread safe I think this function will create the issue.
Imagine calling GetStream on thread safe implementation of this class and
then invoking another thread that uses this stream in non thread safe
manner. Question is can we just give this warning and leave it there or can
we do something more elegant.

Dhruv

On Tue, Nov 10, 2015 at 8:20 AM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/IDataReader.cs
> <https://github.com/apache/incubator-reef/pull/623#discussion_r44426985>:
>
> > @@ -143,5 +144,12 @@ public interface IDataReader
> >          /// <returns>Task handler that reads bytes and returns that number of bytes read
> >          /// if success, otherwise -1</returns>
> >          Task<int> ReadAsync(byte[] buffer, int index, int bytesToRead, CancellationToken token);
> > +
> > +        /// <summary>
> > +        /// Gets underlying stream. Throws null exception if
> > +        /// stream is null or not available
> > +        /// </summary>
> > +        /// <returns>The underlying stream</returns>
> > +        Stream GetStream();
>
> Is this safe if the other methods in this interface are also used?
> @anupam128 <https://github.com/anupam128> Can you have a look please?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/623/files#r44426985>.
>